ironmon_tracker/Main.lua

```diff
@@ -428,17 +432,21 @@ end
 -- Copies 'filename' to 'nameOfCopy' with option to overwrite the file if it exists, or append to it
 -- overwriteOrAppend: 'overwrite' replaces any existing file, 'append' adds to it instead, otherwise no change if file already exists
 function Main.CopyFile(filename, nameOfCopy, overwriteOrAppend)
+	if filename == nil or filename == "" then
+		return false
+	end
+
 	local originalFile = io.open(filename, "rb")
 	if originalFile == nil then
-		print(string.format("Error: Unable to create a backup copy of %s, file doesn't exist."), filename)
+		-- The originalFile to copy doesn't exist, simply do nothing and don't copy
 		return false
 	end
 
 	nameOfCopy = nameOfCopy or (filename .. " (Copy)")
 
 	-- If the file exists but the option to overwrite/append was not specified, avoid altering the file
 	if Main.FileExists(nameOfCopy) and not (overwriteOrAppend == "overwrite" or overwriteOrAppend == "append") then
-		print(string.format("Error: Unable to modify file %s, no overwrite/append option specified."), nameOfCopy)
+		print(string.format('Error: Unable to modify file "%s", no overwrite/append option specified.'), nameOfCopy or "N/A")
```

Comment:
The closing bracket for the string.format is in the wrong place, so the second argument is actually going to the print function here

Same issue in lines 433 (deleted) and 462

Reply:
Ah, that explains so much. I had assumed a different problem. Good catch!